zulipterminal/ui_tools/boxes.py

```diff
@@ -650,21 +704,12 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                     else:
                         header.focus_col = self.FOCUS_HEADER_BOX_STREAM
                 else:
-                    self.update_recipient_emails(self.to_write_box)
-                    invalid_emails = [
-                        recipient_email
-                        for recipient_email in self.recipient_emails
-                        if not self.model.is_valid_private_recipient(
-                            recipient_email
-                        )
-                    ]
-
-                    if invalid_emails:
-                        invalid_emails_error = (
-                            f"Invalid recipient(s) - {', '.join(invalid_emails)}"
-                        )
-                        self.view.controller.report_error(invalid_emails_error)
```

Comment:
Other than adding the feature for the other keypresses, this commit also extracts the code into a function *and* changes it - those could easily be a separate refactor followed by the extra checks.